go-controller/pkg/ovndbmanager/ovndbmanager.go

```diff
@@ -0,0 +1,214 @@
+	for _, member := range members {
+		if len(member) < 2 {
+			klog.Warningf("Unable to find server id submatch in %s", member)
+			return
+		}
+		if member[1] != serverID {
+			// stale entry found for this node with same adddress, need to kick
+			klog.Infof("Previous stale member found: %s...kicking", member[1])
+			_, stderr, err = appCtl("cluster/kick", dbName, member[1])
+			if err != nil {
+				klog.Errorf("Error while kicking old Raft member: %s, for address: %s in db: %s,"+
+					"stderr: %v, error: %v", member[1], addr, db, stderr, err)
+			}
+		}
+	}
+}
+
+// ensureClusterRaftMembership ensures there are no unknown members in the current Raft cluster
+func ensureClusterRaftMembership(db string) {
+	var knownMembers, knownServers []string
+
+	var dbName string
+	var appCtl func(args ...string) (string, string, error)
+
+	if strings.Contains(db, "ovnnb") {
+		dbName = "OVN_Northbound"
+		appCtl = util.RunOVNNBAppCtl
+		knownMembers = strings.Split(config.OvnNorth.Address, ",")
+	} else {
+		dbName = "OVN_Southbound"
+		appCtl = util.RunOVNSBAppCtl
+		knownMembers = strings.Split(config.OvnSouth.Address, ",")
+	}
+	for _, knownMember := range knownMembers {
+		server := strings.Split(knownMember, ":")
+		if len(server) < 3 {
+			klog.Warningf("Failed to parse known member: %s", knownMember)
+			continue
+		}
+		knownServers = append(knownServers, server[1])
+	}
+	out, stderr, err := appCtl("cluster/status", dbName)
+	if err != nil {
+		klog.Warningf("Unable to get cluster status for: %s, stderr: %v, err: %v", db, stderr, err)
+		return
+	}
+	r, _ := regexp.Compile(`([a-z0-9]{4}) at ((ssl|tcp):\[?[a-z0-9.:]+\]?)`)
+	members := r.FindAllStringSubmatch(out, -1)
+	for _, member := range members {
+		if len(member) < 3 {
+			klog.Warningf("Unable to find parse member: %s", member)
+			return
+		}
+		matchedServer := strings.Split(member[2], ":")
+		if len(matchedServer) < 3 {
+			klog.Warningf("Unable to parse address portion of the member entry: %s", matchedServer)
+			return
+		}
+		memberFound := false
+		for _, knownServer := range knownServers {
+			if knownServer == matchedServer[1] {
+				memberFound = true
+				break
+			}
+		}
+		if !memberFound && len(members) > 3 {
+			// unknown member and we have enough members its safe to kick the unknown address
+			klog.Infof("Unknown Raft member found: %s, %s...kicking", member[1], member[2])
+			_, stderr, err = appCtl("cluster/kick", dbName, member[1])
+			if err != nil {
+				// warn only: we might fail to kick since other nodes will also be trying to kick the member
+				klog.Warningf("Error while kicking old Raft member: %s, for address: %s in db: %s,"+
+					"stderr: %v, err: %v", member[1], member[2], db, stderr, err)
+			}
+		}
+	}
+}
+
+// ensureDBHealth ensures that if the db is clustered, it is healthy by calling the cluster-check command
+// if the clustered db is corrupt, the db will be deleted and kube-master pod needs to be started again.
```

Comment:
in case of all db members get corrupted, what will happen? This logic will delete DB for every member, and each DB pod restarts (possibly at different time). When the first DB pod restart, will cluster_exists() check fail or succeed? if the check succeeds and it tries to rejoin/resync from the existing cluster, that would be a problem.

Reply:
So the logic I have here is predicated on the fact that when a new local db instance is created for each member, if the db is truly corrupted, the local instance sync will not happen (it is probably worth confirming that with the ovn team). 

So as they cycle thru' each instance getting deleted and re-created, we should have all members essentially sitting with empty dbs waiting for the ovn-k master to sync them.

Reply:
what we need to check is that when local db is corrupted, the attempt of another node trying to connect to it and do "list-connection --no-leader-only" will fail. Otherwise, this other code will simply trying to rejoin the old cluster.

Reply:
Yes I am checking with the OVN team. @dceara, @numansiddique can you guys comment?

Reply:
All dbs getting corrupted is a use-case that will not be supported by the PR anymore.